lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -22,6 +22,17 @@ using namespace firrtl;
 
 using mlir::TypeStorageAllocator;
 
+static ParseResult parseFIRRTLType(FIRRTLType &result,
+                                   DialectAsmParser &parser);
+
+//===----------------------------------------------------------------------===//
+// TableGen generated logic.
+//===----------------------------------------------------------------------===//
+
+// Provide the autogenerated implementation for types.
+#define GET_TYPEDEF_CLASSES
+#include "circt/Dialect/FIRRTL/FIRRTLTypes.cpp.inc"
```

Comment:
Is it normal to pull in the whole tablegen-generated sources?

Reply:
Yeah, this is the way. It defines some static functions that we have to call like  `generatedTypePrinter(...)`.